tests/Extensions/Hosting.Services.UnitTests/OmexHostedServiceTests.cs

```diff
@@ -0,0 +1,144 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Abstractions;
+using Microsoft.Omex.Extensions.Hosting.Services;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace Hosting.Services.UnitTests
+{
+	[TestClass]
+	public class OmexHostedServiceTests
+	{
+		[TestMethod]
+		public async Task StartAsync_ProperlyCanceled()
+		{
+			MockRunner runnerMock = new MockRunner(t => Task.Delay(int.MaxValue, t));
+			MockLifetime lifetimeMock = new MockLifetime();
+			ILogger<OmexHostedService> loggerMock = new NullLogger<OmexHostedService>();
+
+			OmexHostedService hostedService = new OmexHostedService(runnerMock, lifetimeMock, loggerMock);
+
+			Assert.IsFalse(runnerMock.IsStarted, "RunServiceAsync should not be called after constructor");
+
+			await hostedService.StartAsync(CancellationToken.None);
+
+			Assert.IsFalse(runnerMock.IsStarted, "RunServiceAsync should not be called after StartAsync");
+
+			lifetimeMock.ApplicationStartedSource.Cancel();
+
+			Assert.IsTrue(runnerMock.IsStarted, "RunServiceAsync should not be called after StartAsync");
+
+			Task task = runnerMock.Task!;
+
+			Assert.IsFalse(task.IsCanceled, "Task should not be canceled");
+			Assert.IsFalse(runnerMock.Token.IsCancellationRequested, "CancelationToken should not be canceled");
+
+			lifetimeMock.ApplicationStoppingSource.Cancel();
+
+			Assert.IsTrue(runnerMock.Token.IsCancellationRequested, "Task should be canceled");
+			Assert.IsTrue(task.IsCanceled, "CancelationToken should be canceled");
+		}
+
+
+		[TestMethod]
+		public async Task StartAsync_HandlesExceptions()
+		{
+			MockRunner runnerMock = new MockRunner(t => Task.Run(() => throw new Exception("Totaly valid exeption")));
+			MockLifetime lifetimeMock = new MockLifetime();
+			ILogger<OmexHostedService> loggerMock = new NullLogger<OmexHostedService>();
+
+			OmexHostedService hostedService = new OmexHostedService(runnerMock, lifetimeMock, loggerMock);
+
+			Assert.IsFalse(runnerMock.IsStarted, "RunServiceAsync should not be called after constructor");
+
+			await hostedService.StartAsync(CancellationToken.None);
```

Comment:
If you're using `.ConfigureAwait(false)` within the product code (which is good), you should really use it within the unit tests too to ensure everything aligns.

Adding `.ConfigureAwait(true)` (i.e. the default setting in .NET Framework) means the method will work in far more cases that when `.ConfigureAwait(false)` is present

Reply:
I will add it in the follow-up pr https://github.com/microsoft/Omex/pull/168